webpack/assets/javascripts/react_app/components/AutoCompleteSearch/AutoCompleteSearchBox.js

```diff
@@ -0,0 +1,117 @@
+import React from 'react';
+import _ from 'lodash';
+import { TypeAheadSelect } from 'patternfly-react';
+import { Menu, MenuItem, Highlighter } from 'react-bootstrap-typeahead';
+import './auto-complete.scss';
+import { STATUS } from '../../constants';
+
+class AutoCompleteSearchBox extends React.Component {
+  constructor(props) {
+    super(props);
+    this.handleInputChange = _.debounce(this.handleInputChange, 250);
+    this.handleInputFocus = _.debounce(this.handleInputFocus, 250);
+    this.handleClear = this.handleClear.bind(this);
+    this.filterResult = this.filterResult.bind(this);
+    this.groupBy = this.groupBy.bind(this);
+    this.handleInputChange = this.handleInputChange.bind(this);
+    this.handleItemSelect = this.handleItemSelect.bind(this);
+    this.handleInputFocus = this.handleInputFocus.bind(this);
+    this.handleClear = this.handleClear.bind(this);
+    this.renderMenu = this.renderMenu.bind(this);
```

Comment:
can we use this [style](https://github.com/theforeman/foreman/pull/5771/files#diff-315e6a5832a6d4bbc35c97b46b953729R13)  instead?